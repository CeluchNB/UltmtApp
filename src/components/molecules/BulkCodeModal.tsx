import * as React from 'react'
import Clipboard from '@react-native-clipboard/clipboard'
import { IconButton } from 'react-native-paper'
import PrimaryButton from '../atoms/PrimaryButton'
import { useColors } from '../../hooks'
import { Modal, StyleSheet, Text, View } from 'react-native'
import { size, weight } from '../../theme/fonts'

interface BulkCodeModalProps {
    code: string
    error: string
    visible: boolean
    onClose: () => void
}
const BulkCodeModal: React.FC<BulkCodeModalProps> = ({
    code,
    error,
    visible,
    onClose,
}) => {
    const { colors } = useColors()

    const styles = StyleSheet.create({
        codeIntro: {
            fontSize: size.fontFifteen,
            color: colors.textSecondary,
            fontWeight: weight.full,
            alignSelf: 'center',
        },
        code: {
            fontSize: size.fontLarge,
            letterSpacing: 2,
            color: colors.textSecondary,
            fontWeight: weight.full,
            marginBottom: 10,
        },
        codeContainer: {
            flexDirection: 'row',
        },
        copyButton: {
            flex: 0,
        },
        error: {
            fontSize: size.fontFifteen,
            color: colors.error,
            marginBottom: 10,
        },
        modalContainer: {
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center',
            marginTop: 30,
        },
        modalView: {
            margin: 20,
            backgroundColor: colors.darkGray,
            borderRadius: 10,
            padding: 25,
            alignItems: 'center',
            shadowColor: colors.darkPrimary,
            shadowOffset: {
                width: 5,
                height: 5,
            },
            shadowOpacity: 0.25,
            shadowRadius: 4,
            elevation: 5,
        },
    })

    const copyCode = () => {
        Clipboard.setString(code)
    }

    return (
        <Modal
            animationType="slide"
            transparent={true}
            visible={visible}
            onRequestClose={onClose}
            testID="bulk-code-modal">
            <View style={styles.modalContainer}>
                <View style={styles.modalView}>
                    {code.length > 0 && (
                        <View>
                            <Text style={styles.codeIntro}>Your code is</Text>
                            <View style={styles.codeContainer}>
                                <Text style={styles.code}>{code}</Text>
                                <IconButton
                                    color={colors.darkPrimary}
                                    style={styles.copyButton}
                                    onPress={copyCode}
                                    icon="content-copy"
                                    testID="create-button"
                                />
                            </View>
                        </View>
                    )}
                    {error.length > 0 && (
                        <Text style={styles.error}>{error}</Text>
                    )}
                    <PrimaryButton
                        text="done"
                        loading={false}
                        onPress={async () => {
                            onClose()
                        }}
                    />
                </View>
            </View>
        </Modal>
    )
}

export default BulkCodeModal
