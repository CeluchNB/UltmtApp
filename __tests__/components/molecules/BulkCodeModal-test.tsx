import * as React from 'react'
import BulkCodeModal from '../../../src/components/molecules/BulkCodeModal'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')

it('should match snapshot when visible', async () => {
    const snapshot = renderer.create(
        <BulkCodeModal
            code="123456"
            error=""
            visible={true}
            onClose={() => {}}
        />,
    )

    expect(snapshot).toMatchSnapshot()
})

it('should match snapshot with error', async () => {
    const snapshot = renderer.create(
        <BulkCodeModal
            code=""
            error="Bulk code error"
            visible={true}
            onClose={() => {}}
        />,
    )

    expect(snapshot).toMatchSnapshot()
})

it('should handle close click', async () => {
    const spy = jest.fn()
    const { getByText } = render(
        <BulkCodeModal
            code=""
            error="Bulk code error"
            visible={true}
            onClose={spy}
        />,
    )

    const button = getByText('done')
    fireEvent.press(button)

    expect(spy).toHaveBeenCalled()
})
