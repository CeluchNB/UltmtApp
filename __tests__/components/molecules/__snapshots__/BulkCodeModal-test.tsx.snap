// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when visible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  testID="bulk-code-modal"
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ebebeb",
          "borderRadius": 10,
          "elevation": 5,
          "margin": 20,
          "padding": 25,
          "shadowColor": "#2a2a2a",
          "shadowOffset": Object {
            "height": 5,
            "width": 5,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#31adff",
              "fontSize": 30.821917808219176,
              "fontWeight": "900",
            }
          }
        >
          Your code is
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#31adff",
                "fontSize": 61.64383561643835,
                "fontWeight": "900",
                "letterSpacing": 2,
                "marginBottom": 10,
              }
            }
          >
            123456
          </Text>
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "margin": 6,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 18,
                    "height": 36,
                    "width": 36,
                  },
                  undefined,
                  Object {
                    "flex": 0,
                  },
                ],
              ]
            }
            testID="create-button"
          >
            <View>
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#2a2a2a",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "lineHeight": 24,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                Û∞Üè
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#ffd600",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "minWidth": "auto",
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              done
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`should match snapshot with error 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  testID="bulk-code-modal"
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ebebeb",
          "borderRadius": 10,
          "elevation": 5,
          "margin": 20,
          "padding": 25,
          "shadowColor": "#2a2a2a",
          "shadowOffset": Object {
            "height": 5,
            "width": 5,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ff0000",
            "fontSize": 30.821917808219176,
            "marginBottom": 10,
          }
        }
      >
        Bulk code error
      </Text>
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#ffd600",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "minWidth": "auto",
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              done
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
